### PR TITLE
feat(sdk-coin-ada): add getfee to transaction class

### PR DESCRIPTION
**Pull request by alia-bitgo**
Add transaction fee and getFee to transaction class. The transaction fee and getFee were then de-duplicated from the transaction builder class. Tests were updated accordingly to reflect changes. 

TICKET: [BG-55867](https://bitgoinc.atlassian.net/browse/BG-55867?atlOrigin=eyJpIjoiNzhlNjIyMjFjZTUxNDlhYjkxZjFhYmY3YWExYzA2YTkiLCJwIjoiaiJ9)

### CODE REVIEW

File: modules/sdk-coin-ada/src/lib/transaction.ts [hunk 1]
```diff
@@ -37,6 +38,8 @@ export interface TxData {
 
 export class Transaction extends BaseTransaction {
   private _transaction: CardanoWasm.Transaction;
+  private _fee: BigNum;
+  private _feeSetFromRawTransaction: boolean;
```
Comment (margueriteblair) [inline modules/sdk-coin-ada/src/lib/transaction.ts:42]: ah, the old boolean

File: modules/sdk-coin-ada/src/lib/transactionBuilder.ts [hunk 2]
```diff
@@ -169,9 +168,10 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     );
 
     // calculate the fee based off our dummy transaction
-    if (!this._fee) {
+    if (!this._transaction.getFee || this._transaction.feeSetFromRawTransaction) {
       const fee = CardanoWasm.min_fee(txDraft, linearFee).checked_add(BigNum.from_str('440'));
-      this._fee = fee;
+      this._transaction.fee(fee);
```
Comment (margueriteblair) [inline modules/sdk-coin-ada/src/lib/transactionBuilder.ts:173]: if we're replacing all of the TransactionBuilder.fee instances with Transaction.fee, maybe we can just get rid of that property in TransactionBuilder

File: modules/sdk-coin-ada/test/unit/transactionBuilder.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import * as testData from '../resources';
 import { TransactionBuilderFactory, KeyPair } from '../../src';
 import { coins } from '@bitgo/statics';
 import * as CardanoWasm from '@emurgo/cardano-serialization-lib-nodejs';
+import { Transaction } from '../../src/lib/transaction';
```
Comment (margueriteblair) [inline modules/sdk-coin-ada/test/unit/transactionBuilder.ts:7]: Did we test actually submitting a transaction with Koios for this?

File: modules/sdk-coin-ada/test/unit/transactionBuilder.ts [hunk 4]
```diff
@@ -62,8 +63,8 @@ describe('ADA Transaction Builder', async () => {
     txBuilder.changeAddress(testData.address.address2, '1000000000');
     txBuilder.ttl(800000000);
     txBuilder.sign({ key: testData.privateKeys.prvKey4 });
-    await txBuilder.build();
-    txBuilder.getFee.should.equal('168405');
+    const builtTx = (await txBuilder.build()) as Transaction;
```
Comment (margueriteblair) [inline modules/sdk-coin-ada/test/unit/transactionBuilder.ts:66]: i luv u typescript queen

**Review by margueriteblair — APPROVED** [review]
lgtm 👯‍♀️

**Review by ewangbitgo — APPROVED** [review]

File: modules/sdk-coin-ada/test/unit/transactionBuilder.ts (hunk 3 above)
Comment (alia-bitgo) [reply modules/sdk-coin-ada/test/unit/transactionBuilder.ts:7]: Added in :)

**Review by byhow — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**